### ISSUE TITLE
`Style/SafeNavigation` false negative on method conditionally executed after `.nil?` check

### ISSUE THREAD

**Issue by Zopolis4** [rubocop/rubocop#15086]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False negative

### Cop name

Style/SafeNavigation

### Steps to reproduce the problem

(spotted in the wild as `instance_variable_set("@#{d_pkg}_graph", instance_variable_get("@#{d_pkg}_graph").merge(instance_variable_get("@#{d}_graph"))) unless instance_variable_get("@#{d}_graph").dependencies.nil?`)

Simplified example:
```ruby
bar unless !foo.nil? && !foo.keys.nil?
```

### Expected behavior

To recognize that safe navigation can be applied here and autocorrect to the following:
```ruby
bar unless !foo&.keys.nil?
```

(and then `Style/NegatedUnless` would presumably further correct it to `if foo&.keys.nil?`)

### Actual behavior

No offenses were registered.

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
```

### RuboCop version

```shell
1.86.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.3.8) [x86_64-linux-gnu]
```

**Comment by hammadxcm** [comment]
I took a look at this and wanted to share what I found — there are three independent guards in `on_and` that prevent detection of this pattern. Wanted to get maintainer input on the approach before implementing.

**Guard 1: `ConvertCodeThatCanStartToReturnNil` (default: `false`)**

The `on_and` handler (line 171) skips `!foo.nil? && ...` patterns when this config is `false`:
```ruby
next if !cop_config['ConvertCodeThatCanStartToReturnNil'] && lhs_not_nil_check
```
In the general case `!foo.nil? && foo.bar` → `foo&.bar`, the return value changes from `false` to `nil` when `foo` is nil, so this guard makes sense. But in the reporter's specific pattern (`!foo.nil? && !foo.keys.nil?`), the return value is preserved because `nil.nil?` is `true`, so `!nil&.keys.nil?` still returns `false` when `foo` is nil.

**Guard 2: `negated?` check**

`unsafe_method_used?` calls `negated?`, which walks up from `foo` through `foo.keys` → `foo.keys.nil?` → `!foo.keys.nil?` and finds the `!` operator. This marks the entire chain as unsafe (line 380), causing `offending_node?` to return `false`.

**Guard 3: Chain length**

The chain from `foo` to `!foo.keys.nil?` counts as length 3 (`foo` → `.keys` → `.nil?` → `!`), exceeding the default `MaxChainLength` of 2. The `!` unary method call contributes to the chain length even though it wouldn't need safe navigation.

**Possible approaches:**

1. **Narrow fix:** Special-case the pattern where both sides of `&&` are `!something.nil?` checks on the same receiver chain — this is always safe to convert regardless of `ConvertCodeThatCanStartToReturnNil`.
2. **Context-aware fix:** Teach `on_and` to recognize when the `&&` is used in a conditional context (inside `if`/`unless`), where only truthiness matters and nil-vs-false is irrelevant. This would be more general but higher risk.
3. **Adjust `negated?` / chain length:** Stop counting `!` as part of the chain length when it's the outermost call, and don't mark `.nil?` → `!` as "negated" for unsafe purposes.

Each approach has different scope and risk. What direction would the maintainers prefer?
